### PR TITLE
TSPS-298 Begin GCP migration

### PR DESCRIPTION
**Pull request by mmorgantaylor**
### Description 

This PR cuts over our APIs from being based on Azure to the beginning of using Google. We don't yet have much working in GCP so the new flight is pretty empty and won't succeed at this time, but with this PR, none of our endpoints should touch anything Azure-related.

Specifically:
- modify prepare api and associated methods to not fetch sas urls; for now the prepare endpoint simply returns the inputs the user provided
- modify get results api and associated methods to not fetch sas urls; for now the results endpoint simply returns whatever output values are stored in the pipeline runs db
- start pipeline run creates a GCP imputation flight rather than an Azure imputation flight. the GCP imputation flight only has one tasks (prepare inputs); more will be added in future work
- stop e2e test from running
- updated default value for storage workspace storage container url to the dev storage workspace bucket

Also updated the Google secret for dev and qa containing the Storage workspace storage url to point to the dev workspace bucket:
```
 ✗ gcloud secrets versions access latest --secret="teaspoons-imputation-storage-workspace" --project="broad-dsde-dev"
{"storageUrl": "gs://fc-secure-10efd4d7-392a-4e9e-89ea-d6629fbb06cc"}
```
(same for qa)

### Jira Ticket
https://broadworkbench.atlassian.net/browse/TSPS-298

### CODE REVIEW

File: service/src/main/java/bio/terra/pipelines/stairway/imputation/RunImputationGcpJobFlight.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        RunImputationJobFlightMapKeys.WDL_METHOD_NAME,
+        JobMapKeys.RESULT_PATH.getKeyName());
+
+    PipelinesEnum pipelinesEnum =
+        PipelinesEnum.valueOf(
+            inputParameters.get(JobMapKeys.PIPELINE_NAME.getKeyName(), String.class));
+    MetricsUtils.incrementPipelineRun(pipelinesEnum);
+
+    addStep(
+        new PrepareImputationInputsStep(
```
Comment (jsotobroad) [inline service/src/main/java/bio/terra/pipelines/stairway/imputation/RunImputationGcpJobFlight.java:50]: do we expect this step to work as is or is this just a placeholder for now?

File: service/src/main/java/bio/terra/pipelines/stairway/imputation/CheckLeonardoHealthStep.java [hunk 2]
```diff
@@ -29,7 +29,7 @@ public StepResult doStep(FlightContext flightContext) {
 
   @Override
   public StepResult undoStep(FlightContext flightContext) {
-    // this is the first step in RunImputationJobFlight.
+    // this is the first step in RunImputationAzureJobFlight.
     // increment pipeline failed counter if undoStep is called which means the flight failed
```
Comment (jsotobroad) [inline service/src/main/java/bio/terra/pipelines/stairway/imputation/CheckLeonardoHealthStep.java:33]: we should make sure to add a note somewhere for us to increment this metric in th first step of the gcp flight

**Review by jsotobroad — APPROVED** [review]
some small questions

File: service/src/main/java/bio/terra/pipelines/stairway/imputation/CheckLeonardoHealthStep.java (hunk 2 above)
Comment (mmorgantaylor) [reply service/src/main/java/bio/terra/pipelines/stairway/imputation/CheckLeonardoHealthStep.java:33]: good idea

File: service/src/main/java/bio/terra/pipelines/stairway/imputation/RunImputationGcpJobFlight.java (hunk 1 above)
Comment (mmorgantaylor) [reply service/src/main/java/bio/terra/pipelines/stairway/imputation/RunImputationGcpJobFlight.java:50]: yes, it should work as-is. though that reminds me we also need to update the storage workspace url secret

File: service/src/main/java/bio/terra/pipelines/stairway/imputation/CheckLeonardoHealthStep.java (hunk 2 above)
Comment (mmorgantaylor) [reply service/src/main/java/bio/terra/pipelines/stairway/imputation/CheckLeonardoHealthStep.java:33]: or we implement the webhook ticket, right?

Comment (mmorgantaylor) [reply service/src/main/java/bio/terra/pipelines/stairway/imputation/CheckLeonardoHealthStep.java:33]: I'll make a ticket to add this to the first step of the GCP flight, that's lower lift

Comment (mmorgantaylor) [reply service/src/main/java/bio/terra/pipelines/stairway/imputation/CheckLeonardoHealthStep.java:33]: https://broadworkbench.atlassian.net/browse/TSPS-299

File: service/src/main/java/bio/terra/pipelines/stairway/imputation/RunImputationGcpJobFlight.java (hunk 1 above)
Comment (mmorgantaylor) [reply service/src/main/java/bio/terra/pipelines/stairway/imputation/RunImputationGcpJobFlight.java:50]: actually had to add back putting in the control workspace storage url to the pipeline run. it's hardcoded for now...

Comment (jsotobroad) [reply service/src/main/java/bio/terra/pipelines/stairway/imputation/RunImputationGcpJobFlight.java:50]: can you make a ticket to do update the storage workspace url as well.  I think currently the application config has a default that points to azure

File: service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java [hunk 3]
```diff
@@ -111,88 +95,23 @@ public Map<String, Map<String, String>> preparePipelineRun(
...
-    // return a map of SAS urls and azcopy commands for the user to upload their input files
-    Map<String, Map<String, String>> pipelineFileInputs =
-        prepareFileInputs(pipeline, jobId, userProvidedInputs);
-
-    // get an arbitrary file input (since all file inputs are in the same workspace) and
-    // extract its sasUrl to get the workspace storage URL
-    String arbitraryInputFileSasUrl = pipelineFileInputs.values().iterator().next().get("sasUrl");
-    String workspaceStorageContainerUrl =
-        getStorageContainerUrlFromSasUrl(arbitraryInputFileSasUrl, pipeline.getWorkspaceId());
+    // hardcode for now, ultimately fetch from Rawls or store in config?
```
Comment (mmorgantaylor) [inline service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java:98]: ticket: https://broadworkbench.atlassian.net/browse/TSPS-300

File: service/src/main/java/bio/terra/pipelines/stairway/imputation/RunImputationGcpJobFlight.java (hunk 1 above)
Comment (mmorgantaylor) [reply service/src/main/java/bio/terra/pipelines/stairway/imputation/RunImputationGcpJobFlight.java:50]: just went ahead and did that here, thanks for the reminder

**Pull request merged**